### PR TITLE
newJNL - MIE integration

### PR DESCRIPTION
**Pull request by HakonRydland**
## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [x] Enhancement

### What have you changed and why?
Information:
    Addapting the new JNL system for MIE

### Please specify which Issue this PR Resolves.
closes #XXXX

### Please verify the following and ensure all checks are completed.

1. [ ] Have you loaded the mission in singleplayer?
2. [x] Have you loaded the mission in LAN host?
3. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:

### CODE REVIEW

**Review by SonorousMeerkat — APPROVED** [review]
Was present during Hakon's testing, all seemed fine once a problem with MIE was resolved.

**Comment by CalebSerafin** [comment]
* [ ] Indentation & Headers
* [ ] Naming Conventions & Grammar
* [ ] Macros
* [ ] Param & Call
* [ ] Global Variables
* [ ] Local Variables
* [ ] Logic

File: A3-Antistasi/Templates/3CB/3CB_Logistics_Nodes.sqf [hunk 1]
```diff
@@ -0,0 +1,175 @@
+//JNL mounting nodes for cargo and statics.
+//Each element is: [model name, [nodes]]
+//Nodes are build like this: [Available(internal use,  always 1), Hardpoint location, Seats locked when node is in use]
+A3A_logistics_vehicleHardpoints append [
+  // Datsun civ variant, TODO: redo for 3 crates maybeeee?
+  ["UK3CB_C_Datsun_Open" call A3A_fnc_classNameToModel, [
+      [1,               [0,-0.5,-0.83],     [2,3]],
```
Comment (CalebSerafin) [inline A3-Antistasi/Templates/3CB/3CB_Logistics_Nodes.sqf:7]: 1- The indentation here is level 3 (6 spaces), while the indentaion of it's parent scope is level 1 (2 spaces)
These should be reduced to level 2;

File: A3-Antistasi/functions/Utility/fn_classNameToModel.sqf [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    Arguments:
+    0. <String> Classname
+
+    Return Value:
+    <String> Model path
+
+    Scope: Any
+    Environment: Any
+    Public: Yes
+    Dependencies:
```
Comment (CalebSerafin) [inline A3-Antistasi/functions/Utility/fn_classNameToModel.sqf:17]: 2- You can remove empty header fields.

File: A3-Antistasi/Templates/RHS/RHS_Logistics_Nodes.sqf [hunk 3]
```diff
@@ -0,0 +1,264 @@
+//Each element is: [model name, [nodes]]
+//Nodes are build like this: [Available(internal use,  always 1), Hardpoint location, Seats locked when node is in use]
+A3A_logistics_vehicleHardpoints append [
+    //Urals
+    //Ural Open
+    ["RHS_Ural_Open_Civ_01" call A3A_fnc_classNameToModel, [
+            [1,             [0,0.14,-0.25],         [12,13]],
```
Comment (CalebSerafin) [inline A3-Antistasi/Templates/RHS/RHS_Logistics_Nodes.sqf:7]: 3- Indentaion is at level 3, it should be level 2.

File: A3-Antistasi/functions/Logistics/fn_logistics_initNodes.sqf [hunk 4]
```diff
@@ -1,653 +1,30 @@
 /*
     Author: [Håkon]
     [Description]
-        Defines all vehicle nodes, cargo node, weapon data, blacklist etc. for use with Antistasi logistics system.
-
-        Note: needs to be run on every client for logistic functions to function on said client
+        handles finaliseing of the logistics nodes and distrubuting them to clients
```
Comment (CalebSerafin) [inline A3-Antistasi/functions/Logistics/fn_logistics_initNodes.sqf:4]: 9- Spelling: `finaliseing` should be `finalising`; `distrubuting` should be `distributing`

File: A3-Antistasi/Templates/3CB/3CB_Logistics_Nodes.sqf [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+];
+
+//Offsets for adding new statics/boxes to the JNL script.
+A3A_logistics_attachmentOffset append [];
+
+//all vehicles with jnl loading nodes where the nodes are not located in the open, this can be because its inside the vehicle or it has a cover over the loading plane.
+A3A_logistics_coveredVehicles append ["UK3CB_B_M939_Closed_HIDF", "UK3CB_B_MTVR_Closed_WDL", "UK3CB_C_V3S_Closed"];
+
+//if you want a weapon to be loadable you need to add it to this as a array of [model, [blacklist specific vehicles]],
+//if the vehicle is in the coveredVehicles array dont add it to the blacklist in this array.
```
Comment (CalebSerafin) [inline A3-Antistasi/Templates/3CB/3CB_Logistics_Nodes.sqf:173]: 10- Spelling: `dont` should be `don't` All of the templatse have a spelling mistake at the same position. Remove them before they replicate into new templates.

File: A3-Antistasi/Templates/Example/Example_Logistics_Nodes.sqf [hunk 6]
```diff
@@ -14,18 +13,16 @@ Always think of the next guy that may have to work on your code. Chances are it
 
 /*The first section is for setting the nodes that weapons and boxes should attach to.
 
-    This has the model path for the vehicle, then 3 sections as follows; Node type, Node location and locked seats.
+    This has the model path for the vehicle, then 3 sections as follows; Node size (allways 1), Node location and locked seats.
```
Comment (CalebSerafin) [inline A3-Antistasi/Templates/Example/Example_Logistics_Nodes.sqf:16]: 4- Spelling: `allways` should be `always`

File: A3-Antistasi/Templates/Example/Example_Logistics_Nodes.sqf [hunk 7]
```diff
@@ -14,18 +13,16 @@ Always think of the next guy that may have to work on your code. Chances are it
...
 
-      The node type has 2 values, 0 and 1. 0 is for weapon nodes, i.e. where static turrets should attach to.
-
-      The node location is a set of 3 coordinates in reference to the vehicle model, this defines where the node should be.
+      The node location is a set of 3 coordinates in referance to the vehicle model, this defines where the node should be.
```
Comment (CalebSerafin) [inline A3-Antistasi/Templates/Example/Example_Logistics_Nodes.sqf:18]: 5- Spelling: `referenance` should be `reference`

File: A3-Antistasi/Templates/Example/Example_Logistics_Nodes.sqf [hunk 8]
```diff
@@ -38,36 +35,34 @@ jnl_vehicleHardpoints append [
...
-//Truck
-  ["truckModelPath", jnl_largeVicWeapons]
-];
 
 /*The last section is for defining the offsets for statics, crates and any other item you might want to load onto a vehicle.
   This is usually separated into 3 sections; Weapons, Crates and Other.
   The first 2 are self explanatory, the 3rd is for things like quadbikes, as they can be loaded onto vehicles if they are initialised properly.
-  This is filled by listing the model path, the coordinate offset(for tweaking it so that its base is centered on the node), and any angle offset it needs (in case the weapon should be facing any other direction than forward by default.), finally you list the node type that the entry should use (0 for weapons, 1 for anything else.)
+  This is filled by listing the model path, the coordinate offset(for tweaking it so that its base is centered on the node), and any angle offset it needs (in case the weapon should be facing any other direction than forward by default.),
+  finally you list the node size that the entry should use from 1 to x, for reference a crate usualy is either 1 or 2, and a static is 2 or 4, but it can be any size you want EXCEPT 0 or negative numbers!!!!
```
Comment (CalebSerafin) [inline A3-Antistasi/Templates/Example/Example_Logistics_Nodes.sqf:44]: 6- Spelling: `usualy` should be `usually`

File: A3-Antistasi/Templates/Example/Example_Logistics_Nodes.sqf [hunk 9]
```diff
@@ -38,36 +35,34 @@ jnl_vehicleHardpoints append [
...
+/*
+    Next up is to add all covered or closed vehicles to this next section, this is needed to prevent statics being loaded inside of closed vehicles or covered vehicle, where they become usless, and it quite franckly looks silly.
+*/
+A3A_logistics_coveredVehicles append [
+    "modelPath", "modelPath"
+];
+
+/*
+    Finally you need to declair weapons that you have added here, this is done with arrays consisting of pairs of the model of the weapon, and an array of all vehicle models the weapon is not allowed on.
+    This blacklisted vehicles already include closed and covered vehicles so yuo dont have to add those
```
Comment (CalebSerafin) [inline A3-Antistasi/Templates/Example/Example_Logistics_Nodes.sqf:62]: 7- Spelling: Contraction `dont` should have apostrophy `don't`; `yuo` should be `you`

File: A3-Antistasi/Templates/Example/Example_Logistics_Nodes.sqf [hunk 10]
```diff
@@ -38,36 +35,34 @@ jnl_vehicleHardpoints append [
...
+*/
+A3A_logistics_weapons append [
+    ["weaponModelPath", []]
+    ,["weaponModelPath", ["vehModelPath", "vehModelPath"]]
+];
+//That covers everything, you should make you file by replacing values in an already complete file rather than using this as the active files will have the propper commenting there already. Using this one would leave a tonne of unnecessary comments in the file.
```
Comment (CalebSerafin) [inline A3-Antistasi/Templates/Example/Example_Logistics_Nodes.sqf:68]: 8- Spelling: Proper spelling of `propper` should be `proper`

**Review by CalebSerafin — APPROVED** [review]
Code is solid, couple of indention and spelling concerns.
Here is an [Excel Spreadsheet](https://1drv.ms/x/s!AhRKsW_EtNcb3UHq78KtEi3girbA?e=5IgSAG) of the comments (Valid until 1 Feb)

**Review by Bob-Murphy — APPROVED** [review]

**Pull request merged**